cmd/celestia/blob.go

```diff
@@ -0,0 +1,200 @@
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		height, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a height:%v", err)
+		}
+
+		namespace, err := parseV0Namespace(args[1])
+		if err != nil {
+			return fmt.Errorf("error parsing a namespace:%v", err)
+		}
+
+		var output []byte
+		blobs, err := client.Blob.GetAll(cmd.Context(), height, []share.Namespace{namespace})
+		if err != nil {
+			output, err = prepareOutput(err)
+			if err != nil {
+				return err
+			}
+		} else {
+			output, err = prepareOutput(blobs)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintln(os.Stdout, string(output))
+		return nil
+	},
+}
+
+var submitCmd = &cobra.Command{
+	Use:   "submit [params]",
+	Args:  cobra.ExactArgs(2),
+	Short: "Submit the blob at the given namespace. Note: only one blob is allowed to submit through the RPC.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		namespace, err := parseV0Namespace(args[0])
+		if err != nil {
+			return fmt.Errorf("error parsing a namespace:%v", err)
+		}
+
+		parsedBlob, err := blob.NewBlobV0(namespace, []byte(args[1]))
+		if err != nil {
+			return fmt.Errorf("error creating a blob:%v", err)
+		}
+
+		var output []byte
+		height, err := client.Blob.Submit(cmd.Context(), []*blob.Blob{parsedBlob})
+		if err != nil {
+			output, err = prepareOutput(err)
+			if err != nil {
+				return err
+			}
+		} else {
+			output, err = prepareOutput(height)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintln(os.Stdout, string(output))
+		return nil
+	},
+}
+
+var getProofCmd = &cobra.Command{
+	Use:   "getProof [params]",
+	Args:  cobra.ExactArgs(3),
+	Short: "Retrieves the blob in the given namespaces at the given height by commitment and returns its Proof.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		height, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return fmt.Errorf("error parsing a height:%v", err)
+		}
+
+		namespace, err := parseV0Namespace(args[1])
+		if err != nil {
+			return fmt.Errorf("error parsing a namespace:%v", err)
+		}
+
+		commitment, err := base64.StdEncoding.DecodeString(args[2])
+		if err != nil {
+			return fmt.Errorf("error parsing a commitment:%v", err)
+		}
+
+		var output []byte
+		proof, err := client.Blob.GetProof(cmd.Context(), height, namespace, commitment)
+		if err != nil {
+			output, err = prepareOutput(err)
+			if err != nil {
+				return err
+			}
+		} else {
+			output, err = prepareOutput(proof)
+			if err != nil {
+				return err
+			}
+		}
+
+		fmt.Fprintln(os.Stdout, string(output))
+		return nil
+	},
+}
+
+func prepareOutput(data interface{}) ([]byte, error) {
+	bytes, err := json.MarshalIndent(response{
+		Result: data,
+	}, "", "  ")
```

Comment:
Why wrap in response?

Reply:
We have such approach for all responses mainly because in case of  multiple returned values, you can use indexing over the output. Please check getAll cmd and  pipelining with jq

Reply:
There are standard unix pipes for that. I don't think we should enforce json here

Reply:
Keeping it JSON makes sense, as we all got used to it. In a perfect world, we would have a flag to specify which encoding to use with plaintext by default and option to choose json.